swap/tests/testutils/electrs.rs

```diff
@@ -0,0 +1,164 @@
+
+    fn descriptor(&self) -> String {
+        format!("vulpemventures/electrs:{}", self.tag)
+    }
+
+    fn wait_until_ready<D: Docker>(&self, container: &Container<'_, D, Self>) {
+        container
+            .logs()
+            .stderr
+            .wait_for_message(&self.wait_for_message)
+            .unwrap();
+    }
+
+    fn args(&self) -> <Self as Image>::Args {
+        self.args.clone()
+    }
+
+    fn volumes(&self) -> Self::Volumes {
+        let mut volumes = HashMap::new();
+        volumes.insert(self.volume.clone(), "/home/bdk-test".to_owned());
+        volumes
+    }
+
+    fn env_vars(&self) -> Self::EnvVars {
+        HashMap::new()
+    }
+
+    fn ports(&self) -> Option<Vec<Port>> {
+        self.ports.clone()
+    }
+
+    fn with_args(self, args: <Self as Image>::Args) -> Self {
+        Electrs { args, ..self }
+    }
+
+    fn with_entrypoint(self, entrypoint: &Self::EntryPoint) -> Self {
+        Self {
+            entrypoint: Some(entrypoint.to_string()),
+            ..self
+        }
+    }
+
+    fn entrypoint(&self) -> Option<String> {
+        self.entrypoint.to_owned()
+    }
+}
+
+impl Default for Electrs {
+    fn default() -> Self {
+        Electrs {
+            tag: "v0.16.0.3".into(),
+            args: ElectrsArgs::default(),
+            ports: None,
+            entrypoint: Some("/build/electrs".into()),
+            wait_for_message: "Running accept thread".to_string(),
+            volume: uuid::Uuid::new_v4().to_string(),
+            bitcoind_container_name: uuid::Uuid::new_v4().to_string(),
+        }
+    }
+}
+
+impl Electrs {
+    pub fn with_tag(self, tag_str: &str) -> Self {
+        Electrs {
+            tag: tag_str.to_string(),
+            ..self
+        }
+    }
+
+    pub fn with_mapped_port<P: Into<Port>>(mut self, port: P) -> Self {
+        let mut ports = self.ports.unwrap_or_default();
+        ports.push(port.into());
+        self.ports = Some(ports);
+        self
+    }
+
+    pub fn with_volume(mut self, volume: String) -> Self {
+        self.volume = volume;
+        self
+    }
+
+    pub fn with_daemon_rpc_addr(mut self, name: String) -> Self {
+        self.args.daemon_rpc_addr = name;
+        self
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ElectrsArgs {
+    pub network: Network,
+    pub daemon_dir: String,
+    pub daemon_rpc_addr: String,
+    pub cookie: String,
+    pub http_addr: String,
+    pub electrum_rpc_addr: String,
+    pub cors: String,
+}
+
+/// Sane defaults for a mainnet regtest instance.
+impl Default for ElectrsArgs {
+    fn default() -> Self {
+        ElectrsArgs {
+            network: Network::Regtest,
+            daemon_dir: "/home/bdk-test/".to_string(),
+            daemon_rpc_addr: "0.0.0.0:18443".to_string(),
+            cookie: "admin:123".to_string(),
+            http_addr: "0.0.0.0:3002".to_string(),
+            electrum_rpc_addr: "0.0.0.0:60401".to_string(),
+            cors: "*".to_string(),
```

Comment:
Most of the defaults don't make sense as electrum-rs won't work then.
Afaik, electrum-rs has some smart defaults which would just connect to public bitcoin nodes.

Reply:
Are suggesting that ElectrsArgs should be an empty struct? This image is only used by the tests in these projects. Shouldn't the defaults reflect this?

Reply:
Used constants where I can added a todo to fix this later